File: internal/models/listing.go
package models

import (
	"time"

	"github.com/lib/pq"
)

type Listing struct {
	ID                   int            `db:"id" json:"id"`
	Name                 string         `db:"name" json:"name"`
	Area                 string         `db:"area" json:"area"`
	Address              string         `db:"address" json:"address"`
	UniversityNearby     *string        `db:"university_nearby" json:"university_nearby"`
	PricePerMonth        int            `db:"price_per_month" json:"price_per_month"`
	Description          *string        `db:"description" json:"description"`
	ThumbnailURL         *string        `db:"thumbnail_url" json:"thumbnail_url"`
	PhotoURLs            pq.StringArray `db:"photo_urls" json:"photo_urls"`
	Facilities           pq.StringArray `db:"facilities" json:"facilities"`
	IsAvailable          bool           `db:"is_available" json:"is_available"`
	LandlordWANumber     string         `db:"landlord_wa_number" json:"landlord_wa_number"`
	IsFeatured           bool           `db:"is_featured" json:"is_featured"`
	AverageRating        float64        `db:"average_rating" json:"average_rating"`
	GoogleMapsIframeURL  *string        `db:"google_maps_iframe_url" json:"google_maps_iframe_url"`
	CreatedAt            time.Time      `db:"created_at" json:"created_at"`
	UpdatedAt            time.Time      `db:"updated_at" json:"updated_at"`
}

File: mobile/lib/models/listing.dart
class Listing {
  final int id;
  final String name;
  final String area;
  final String address;
  final String? universityNearby;
  final int pricePerMonth;
  final String? description;
  final String? thumbnailUrl;
  final List<String> photoUrls;
  final List<String> facilities;
  final bool isAvailable;
  final String landlordWaNumber;
  final bool isFeatured;
  final double averageRating;
  final String kosType;
  final String? googleMapsIframeUrl;

  Listing({
    required this.id,
    required this.name,
    required this.area,
    required this.address,
    this.universityNearby,
    required this.pricePerMonth,
    this.description,
    this.thumbnailUrl,
    required this.photoUrls,
    required this.facilities,
    required this.isAvailable,
    required this.landlordWaNumber,
    required this.isFeatured,
    required this.averageRating,
    required this.kosType,
    this.googleMapsIframeUrl,
  });

  factory Listing.fromJson(Map<String, dynamic> json) {
    return Listing(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      area: json['area'] ?? '',
      address: json['address'] ?? '',
      universityNearby: json['university_nearby'],
      pricePerMonth: json['price_per_month'] ?? 0,
      description: json['description'],
      thumbnailUrl: json['thumbnail_url'],
      photoUrls: List<String>.from(json['photo_urls'] ?? []),
      facilities: List<String>.from(json['facilities'] ?? []),
      isAvailable: json['is_available'] ?? false,
      landlordWaNumber: json['landlord_wa_number'] ?? '',
      isFeatured: json['is_featured'] ?? false,
      averageRating: (json['average_rating'] ?? 0).toDouble(),
      kosType: json['kos_type'] ?? 'campur',
      googleMapsIframeUrl: json['google_maps_iframe_url'],
    );
  }
}
